Solutions/Cloud Identity Threat Protection Essentials/Hunting Queries/Signins-from-NordVPN-Providers.yaml

```diff
@@ -23,7 +29,35 @@ query: |
     | join kind= inner nord_vpn_feed on $left.IPAddress == $right.ip_address
     | project StartTime , EndTime, IPAddress, UserPrincipalName, AppList, ClientAppUsed, ConditionalAccessStatus, AuthenticationRequirement, RiskDetail, categories, domain, country
     | extend timestamp = StartTime, AccountCustomEntity = UserPrincipalName, IPCustomEntity = IPAddress
-
+    | join kind=leftouter (
+        IdentityInfo
+        | summarize LatestReportTime = arg_max(TimeGenerated, *) by AccountUPN
+        | extend BlastRadiusInt = iif(BlastRadius == "High", 1, 0)
+        | project AccountUPN, Tags, JobTitle, GroupMembership, AssignedRoles, UserType, IsAccountEnabled, BlastRadiusInt
+        | summarize
+            Tags = make_set(Tags),
+            GroupMembership = make_set(GroupMembership),
+            AssignedRoles = make_set(AssignedRoles),
+            BlastRadiusInt = sum(BlastRadiusInt),
+            UserType = make_set(UserType),
+            UserAccountControl = make_set(UserType)
+        by AccountUPN
+        | extend UserPrincipalName=tolower(AccountUPN)
+    ) on UserPrincipalName
+    | join kind=leftouter (
+        BehaviorAnalytics
+        | where ActivityType in ("FailedLogOn", "LogOn")
+        | where isnotempty(SourceIPAddress)
+        | project UsersInsights, DevicesInsights, ActivityInsights, InvestigationPriority, SourceIPAddress
+        | project-rename IPAddress = SourceIPAddress
+        | summarize
+          UsersInsights = make_set(UsersInsights, 1000),
+          DevicesInsights = make_set(DevicesInsights, 1000),
+          IPInvestigationPriority = sum(InvestigationPriority)
+        by IPAddress
+    ) on IPAddress
+    | extend UEBARiskScore = BlastRadiusInt + IPInvestigationPriority
```

Comment:
Given these are looking for VPS providers explicitly could we use the ActivityInsight of {"ISPUncommonlyUsedInTenant":"True"} to explicitly look for VPS that are uncommon?

Reply:
Added!